### PR TITLE
added login functionality

### PR DESCRIPTION
**Pull request by fiyinfoluwaafol**
fixed minor bugs discovered in last PR with users signing up
user is now able to login with their email and password
also implemented session management such that after refreshing page, user remains logged in

Attached is a video walkthrough showing what i implemented

https://www.loom.com/share/8ba49c9ba68148a9959ab018cf86c817?sid=70268240-6088-40c9-8658-76a56de97e52

Next PR should include the user being able to log out

### CODE REVIEW

File: frontend/src/components/LoginForm/LoginForm.jsx [hunk 1]
```diff
@@ -1,21 +1,58 @@
...
     const [password, setPassword] = useState("");
+    const { updateUser } = useContext(UserContext);
+    const navigate = useNavigate()
 
-    function handleOnSubmit (event) {
+
+    const userObj = {
+        email: "",
+        password: ""
+    }
```
Comment (jameskim08) [inline frontend/src/components/LoginForm/LoginForm.jsx:17]: These lines are not necessary.

File: frontend/src/components/LoginForm/LoginForm.jsx [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
         event.preventDefault();
-        event.stopPropagation();
-        const userObj = {
+        if (email && password) {
+          const userObj = {
             email: email,
             password: password
+          };
+
+          console.log(userObj);
```
Comment (jameskim08) [inline frontend/src/components/LoginForm/LoginForm.jsx:26]: remove.

File: frontend/src/components/LoginForm/LoginForm.jsx [hunk 3]
```diff
@@ -1,21 +1,58 @@
...
+            if (!response.ok) {
+              throw new Error('Login failed');
+            }
+
+            const data = await response.json();
+            const { user } = data;
+
+            updateUser(user);
+            navigate('/user/5/calendar');
+            alert("Successfully Logged In");
```
Comment (jameskim08) [inline frontend/src/components/LoginForm/LoginForm.jsx:47]: You don't need an alert for successful logins

File: frontend/src/components/SignUpForm/SignUpForm.jsx [hunk 4]
```diff
@@ -9,7 +9,7 @@ function SignUpForm (){
     const [firstName, setFirstName] = useState("");
     const [lastName, setLastName] = useState("");
 
-    // const { updateUser } = useContext(UserContext);
+    const { updateUser } = useContext(UserContext);
     const navigate = useNavigate();
 
     const handleOnSubmit = async (event) => {
```
Comment (jameskim08) [inline frontend/src/components/SignUpForm/SignUpForm.jsx:15]: You should check that the fields are all not empty before fetching the backend.
Also some validation for password and email would be good.

File: backend/routes/userAuth.js
Comment (juancafe4) [inline backend/routes/userAuth.js:1]: let's called it `Auth.js` should the endpoint be `auth/` e.g .`auth//login`

File: backend/routes/userAuth.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    req.user = user;
+    next();
+  });
+};
+
+// Definition of the routes for user registration, logining in and logging out
+// User Registration/Sign Up
+router.post('/users', async (req, res) => {
```
Comment (juancafe4) [inline backend/routes/userAuth.js:32]: `/registration` makes more sense what do you think?

File: backend/routes/userAuth.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    const {email, password, firstName, lastName} = req.body;
```
Comment (juancafe4) [inline backend/routes/userAuth.js:33]: should we use an email validator like this one? https://www.npmjs.com/package/email-validator

Comment (juancafe4) [reply backend/routes/userAuth.js:33]: I know you might be doing this in the frontend. But it is a good practice to catch errors in the frontend and backend.

File: backend/index.js [hunk 7]
```diff
@@ -1,43 +1,26 @@
...
 
 app.use(express.json());
-app.use(cors());
+app.use(cookieParser());
+app.use(cors({
+  origin: 'http://localhost:5173',
+  credentials: true
+}));
+
+app.use(userRoutes);
```
Comment (juancafe4) [inline backend/index.js:21]: `app.use('auth/', authRoute);`

**Pull request merged**